frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala

```diff
@@ -331,6 +331,53 @@ class BspMetalsClientSpec(
     }
   }
 
+  test("compile with scala3 without using plugin") {
+    TestUtil.withinWorkspace { workspace =>
+      val `A` = TestProject(
+        workspace,
+        "A",
+        dummyFooSources,
+        scalaVersion = Some("0.21.0-RC1"),
+        scalaOrg = Some("ch.epfl.lamp"),
+        scalaCompiler = Some("dotty-compiler_0.21")
+      )
+      val projects = List(`A`)
+      val configDir = TestProject.populateWorkspace(workspace, projects)
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      WorkspaceSettings.writeToFile(
+        configDir,
+        WorkspaceSettings("4.3.0", List()),
+        logger
+      )
+      loadBspState(workspace, projects, logger) { state =>
+        val compiledState = state.compile(`A`).toTestState
+        assert(compiledState.status == ExitStatus.Ok)
```

Comment:
Let's try to add some of the other checks we have in `CompileSpec` for simple hello world compilations in Scala 2. You can find these checks at the beginning of the `CompileSpec` definition.

Reply:
Should we create a separate CompileSpec for Scala 3? Or do you mean to add them here?

Reply:
Or just he asserts from the test cases there?